logutil/logutil_test.go

```diff
@@ -0,0 +1,99 @@
+package logutil_test
+
+import (
+	"bytes"
+	"io"
+	"log/slog"
+	"testing"
+	"time"
+
+	"github.com/TouchBistro/goutils/logutil"
+)
+
+func TestWriterVar(t *testing.T) {
+	var wv logutil.WriterVar
+	data := []byte("hello")
+	// Check that a zero value works
+	if gotN, gotErr := wv.Write([]byte("hello")); gotN != len(data) || gotErr != nil {
+		t.Errorf("got %d, %v; want %d, nil", gotN, gotErr, len(data))
+	}
+	var b bytes.Buffer
+	wv.Set(&b)
+	if gotN, gotErr := wv.Write([]byte("hello")); gotN != len(data) || gotErr != nil {
+		t.Errorf("got %d, %v; want %d, nil", gotN, gotErr, len(data))
+	}
+	if got := b.String(); got != "hello" {
+		t.Errorf("got %q; want %q", got, "hello")
+	}
+}
+
+func TestLogWriter(t *testing.T) {
```

Comment:
What happens if the logger returns an error when you try to write to it? panic?

Reply:
Well the logger API doesn't allow for an error return. There could be extreme cases where it panics but shouldn't really happen in practice